frontend/src/libs/components/pagination/pagination.tsx

```diff
@@ -0,0 +1,146 @@
+import { getValidClassNames } from '~/libs/helpers/helpers.js';
+import { useCallback } from '~/libs/hooks/hooks.js';
+
+import { Button } from '../button/button.jsx';
+import { DEFAULT_SIZE } from './libs/constant.js';
+import { getMiddle } from './libs/helper.js';
+import styles from './styles.module.scss';
+
+type Properties = {
+  pageCount: number;
+  pageIndex: number;
+  pageSize: number;
+  size?: number;
+  onClick: (n: number) => void;
+  onChangePageSize: (n: number) => void;
+};
+
+const Pagination: React.FC<Properties> = ({
+  pageCount,
+  pageIndex,
+  pageSize,
+  size = pageCount > DEFAULT_SIZE ? DEFAULT_SIZE : pageCount,
+  onClick,
+  onChangePageSize,
+}: Properties) => {
+  const middleValue = getMiddle(size);
+
+  const isFirstPage = pageIndex === 0;
+  const isLastPage = pageIndex === pageCount - 1;
+
+  const isHiddenFirstPage =
+    pageIndex >= size - 1 && size < pageCount && pageIndex > 0;
+
+  const isHiddenLastPage =
+    pageIndex + middleValue < pageCount - 1 && size < pageCount;
```

Comment:
Let's create a helper to convert page number to page index? That operation `pageCount -1` 
And opposite to convert page index to page number